### PR TITLE
fix: fast-number-field does not fire input event when buttons pressed

### PR DESCRIPTION
**Pull request by robarbms**
# Description

Firing the input event when using the up and down arrows to change the value.

## Motivation & context

Changing the value using the up and down arrows should fire the input event.

https://github.com/microsoft/fast/issues/4508

## Issue type checklist

- [ ] **Chore**: A change that does not impact distributed packages.
- [x] **Bug fix**: A change that fixes an issue, link to the issue above.
- [ ] **New feature**: A change that adds functionality.

**Is this a breaking change?**
- [ ] This change causes current functionality to break.

**Adding or modifying component(s) in `@microsoft/fast-components` checklist**

- [ ] I have added a new component
- [x] I have modified an existing component
- [ ] I have updated the [definition file](https://github.com/Microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#definition)
- [ ] I have updated the [configuration file](https://github.com/Microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#configuration)

## Process & policy checklist

- [ ] I have added tests for my changes.
- [x] I have tested my changes.
- [ ] I have updated the project documentation to reflect my changes.
- [x] I have read the [CONTRIBUTING](https://github.com/Microsoft/fast/blob/master/CONTRIBUTING.md) documentation and followed the [standards](https://www.fast.design/docs/community/code-of-conduct/#our-standards) for this project.

### CODE REVIEW

**Review by EisenbergEffect — APPROVED** [review]

**Comment by scomea** [comment]
Adding tests for this would be good.

**Review by scomea — APPROVED** [review]

**Review by nicholasrice — APPROVED** [review]

File: packages/web-components/fast-foundation/src/number-field/number-field.spec.ts [hunk 1]
```diff
@@ -519,6 +519,31 @@ describe("NumberField", () => {
 
             await disconnect();
         });
+
+        it("should fire an input event when incramenting or decramenting", async () => {
```
Comment (janechu) [inline packages/web-components/fast-foundation/src/number-field/number-field.spec.ts:523]:
```suggestion
        it("should fire an input event when incrementing or decrementing", async () => {
```

**Review by janechu — APPROVED** [review]

**Comment by chrisdholt** [comment]
@robarms is this good to go?

**Comment by robarbms** [comment]
> @robarms is this good to go?

Yes sir.

**Pull request merged**